pkg/agent/config/config.go

```diff
@@ -365,3 +373,29 @@ func HostnameCheck(cfg cmds.Agent) error {
 	}
 	return fmt.Errorf("Timed out waiting for hostname %s to be resolvable: %v", hostname, err)
 }
+
+func getIPFromInterface(flannelIfaceName string) (string, error) {
+	flannelIface, err := sysnet.InterfaceByName(flannelIfaceName)
+	if err != nil {
+		return "", err
+	}
+	addrs, err := flannelIface.Addrs()
```

Comment:
Can we just sort this list so that it is deterministic which IP is returned.  I don't think the result of Addrs() is always going to be the same order.

Reply:
fixed